patches/server/0966-Add-player-death-screen-message-API.patch

```diff
@@ -0,0 +1,37 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: dawon <dawon@dawon.eu>
+Date: Sun, 12 Mar 2023 17:17:40 +0100
+Subject: [PATCH] Add player death screen message API
+
+
+diff --git a/src/main/java/net/minecraft/server/level/ServerPlayer.java b/src/main/java/net/minecraft/server/level/ServerPlayer.java
+index 289429eb464548acc80262a49444f49f8f57fc0c..e3f98f4930d8167b8c7d57363c9faad87aaacea2 100644
+--- a/src/main/java/net/minecraft/server/level/ServerPlayer.java
++++ b/src/main/java/net/minecraft/server/level/ServerPlayer.java
+@@ -922,13 +922,14 @@ public class ServerPlayer extends Player {
+         }
+ 
+         net.kyori.adventure.text.Component deathMessage = event.deathMessage() != null ? event.deathMessage() : net.kyori.adventure.text.Component.empty(); // Paper - Adventure
++        Component deathScreenMessage = PaperAdventure.asVanilla(event.deathScreenMessage() != null && flag ? event.deathScreenMessage() : net.kyori.adventure.text.Component.empty()); // Paper - death screen api
+ 
+-        if (deathMessage != null && deathMessage != net.kyori.adventure.text.Component.empty() && flag) { // Paper - Adventure // TODO: allow plugins to override?
++        if (deathMessage != null && deathMessage != net.kyori.adventure.text.Component.empty() && flag) { // Paper - Adventure
```

Comment:
Why the removal here ? I don't think that comment related to the death screen api more to the `flag` or well gamerule.

Reply:
Shall I also add an api to turn on/off display death messages as well then?

Reply:
Probably not in this PR.

Reply:
I rebranded the PR & commits names and added API to get/set whether the death message should be shown.